sentry-android-core/src/main/java/io/sentry/android/core/internal/util/ScreenshotUtils.java

```diff
@@ -55,20 +61,57 @@ public class ScreenshotUtils {
       final Bitmap bitmap =
           Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.ARGB_8888);
 
-      final Canvas canvas = new Canvas(bitmap);
-      if (mainThreadChecker.isMainThread()) {
-        view.draw(canvas);
-      } else {
-        final @NotNull CountDownLatch latch = new CountDownLatch(1);
-        activity.runOnUiThread(
-            () -> {
-              try {
-                view.draw(canvas);
+      final @NotNull CountDownLatch latch = new CountDownLatch(1);
+
+      // Use Pixel Copy API on new devices, fallback to canvas rendering on older ones
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
`l`: We could use `BuildInfoProvider` here and test it. You could take a look at `SentryGestureListenerClickTest` where we completely mock windows and decor views, and there's also [ShadowPixelCopy](https://github.com/robolectric/robolectric/blob/b0c04e9b34b66cd300448dd29b8ccc26b22f3f78/shadows/framework/src/main/java/org/robolectric/shadows/ShadowPixelCopy.java#L44) of robolectric that would be helpful

Reply:
Completely forget about that, I've added some test using `ShadowPixelCopy`